app/controllers/caterings_controller.rb

```diff
@@ -12,23 +12,39 @@ def show
 	end
 
 	def new
-		
+		@catering = Catering.new
 	end
 
 	def create
-		
+		@catering = Catering.new(catering_params)
+		if @catering.save
+			redirect_to root_path
+			flash[:notice] = "Catering created"
+		else
+			redirect_to new_catering_path
+			flash[:alert] = "Try again!"
+		end
 	end
 
 	def edit
-
+		@catering = Catering.find_by_id(params[:id])
 	end
 
 	def update
-
+		@catering = Catering.find_by_id(params[:id])
+		if @catering.update(catering_params)
+			redirect_to root_path
+			flash[:notice] = "Catering updated"
+		else
+			redirect_to edit_catering_path
+			flash[:alert] = "Try again!"
+		end
 	end
 
 	def destroy
-
+		@catering = Catering.find_by_id(params[:id])
```

Comment:
You don't really want to destroy caterings. Think of the consequences, broken links in the DB etc.